### ISSUE TITLE
^D doesn't EOF/exit/restart blink shell

### ISSUE THREAD

**Issue by sbromberger** [blinksh/blink#813]
at the `blink>` prompt, `^D` used to terminate and restart the blink shell. It no longer does this. Was this intentional? It was a good trick to see whether network access was available if you used a custom font (the font appears to be loaded dynamically from the internet when the shell starts; if no network is available you get instead the default blink font).

Also, possibly unrelated, typing `exit` at the blink prompt resulted in a complete hang of the app. I am unable to reliably reproduce this one.

Blink 13.0.147 on iPadOS 13.1.2.

**Comment by jmeosbn** [comment]
The removal of ^C also doing this at the same time makes sense so I'm not sure if ^D (in which many shells explicitly enter exit^M) was caught up by mistake.

Having to type exit or three finger tap is frustrating on an iPhone - perhaps not so on an iPad?

**Comment by sbromberger** [comment]
> Having to type exit or three finger tap is frustrating on an iPhone - perhaps not so on an iPad?

It's still frustrating because years of muscle memory have trained me to recognize `^D` as the way to exit a shell.

**Comment by jmeosbn** [comment]
Yes, the ^C had the same problem with muscle memory losing a window when only wanting to clear the command line! 😣 

I think both ^C and ^D exited blink shell even when command line wasn't empty so that was an issue too. (^D is even used in this context by zsh to preview completions).

Jamie Osborne

> On 7 Oct 2019, at 15:41, Seth Bromberger - notifications@github.com <github.jamieo.542923af64.notifications#reply@reply.github.com> wrote:
> 

﻿
Having to type exit or three finger tap is frustrating on an iPhone - perhaps not so on an iPad?

It's still frustrating because years of muscle memory have trained me to recognize ^D as the way to exit a shell.

**Comment by sbromberger** [comment]
> Yes, the ^C had the same problem with muscle memory losing a window when only wanting to clear the command line! 

I can see how that would be frustrating. I guess my preference would be for `^C` to "discard current line" as it does in (most) shells, and `^D` to terminate the shell. (Why *shouldn't* blink support EOF - the closure of standard input - as a signal to exit?)

**Comment by jmeosbn** [comment]
Agree that's pretty standard and expected behaviour - with the addition that `^D` deletes next character in mid line (_does not currently in blink shell*_) and is either ignored or lists completions when at the end of non-empty line (as in bash/zsh).

> given del forward character is also broken while `Alt-D / ^F / ^B` all work I think this was unintentional change when `^C` behaviour was changed. @yury ?

**Comment by jmeosbn** [comment]
Excellent to hear (well, hinted) that repl is getting some love. I really want to use blink for for on device stuff but find it completion spotty (often refusing to complete or other issues with spaces) and lack of aliases a hindrance leading me back to Pythonista.

P.S. I've submitted some issues via TestFlight (as it was convenient for screenshots etc.) but should I open new issues here?

**Comment by sbromberger** [comment]
@yury - any update on this? It's becoming a bit more urgent now that there's another bug (in the beta; I've sent feedback) that causes blink to open 100+ new sessions, and the only way to get rid of them is to type `exit` in each one. Holding down `^D`, like we used to be able to do, would be much easier. Thanks!

**Comment by sbromberger** [comment]
ping on this - having ^C and ^D work as they do in regular shells would be a huge time-saver.
